### DISCUSSION TITLE
internal error dynamic scan

### DISCUSSION THREAD

**Discussion by lin-ycv** [microsoft/winget-pkgs#155218] [Q&A] [answered]
What is `internal error dynamic scan`? My package keeps getting this label and the process would rerun, causing a loop.

**Comment by stephengillie** [comment, marked as answer]
The label `Internal-Error-Dynamic-Scan` is an umbrella term for internal errors encountered during the dynamic scan step of manual validation. This only indicates a pipeline issue and doesn't reflect on the package. Part of my role is to keep track of these and report them to the team, for any trends to be deduced and remediated. 

Frequently, this is an ephemeral issue that will be applied to a PR, then removed on the subsequent run. This might be due to a network issue or other temporary situation. 

Sometimes, it applies to a PR on every run. This is sometimes due to the PR adding a package that's used in the validation process. But it's recently become revealed that sometimes this is due to Storage Drive out of space errors as well, impacting very large packages. For these, another part of my role is to manually install and scan in a VM. (I've automated most of it - I press 1 button at start, watch the VM window's progress behind browser and other windows, and then look at the output.)

**Comment by stephengillie** [comment]
The label https://github.com/microsoft/winget-pkgs/labels/Internal-Error-Static-Scan is very similar to its Dynamic counterpart, described above. The difference is that this Internal Error occurred during the Static Scan phase of Validation, where packages go through static scans such as detecting "installer bombs". This label is rarely seen because this step is much simpler to automate and thus has many fewer errors.
